mathics/builtin/assignments/assignment.py

```diff
@@ -18,6 +37,62 @@
 from mathics.core.systemsymbols import SymbolFailed
 
 
+class _SetOperator:
+    """
+    This is the base class for assignment Builtin operators.
+
+    Special cases are determined by the head of the expression. Then
+    they are processed by specific routines, which are poke from
+    the `special_cases` dict.
+    """
+
+    # FIXME:
+    # Assigment is determined by the LHS.
+    # Is there a larger patterns or natural grouping that we are missing?
+    # For example it might be that
+    # there are some attributes or other properties of of the
+    # LHS of a builtin that we should be targetting.
+    # Below, we key on a string, but Symbol is more correct.
+    #
+
+    special_cases = {
```

Comment:
Maybe this dictionary could be moved also to marhics.core.assignment

Reply:
Good idea! Will do next.